android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiTableView.java

```diff
@@ -194,23 +194,20 @@ public View getView(int position, View convertView, ViewGroup parent) {
 				// Default creates view for each Item
 				boolean sameView = false;
 				if (item.proxy instanceof TableViewRowProxy) {
-					TableViewRowProxy row = (TableViewRowProxy)item.proxy;
+					TableViewRowProxy row = (TableViewRowProxy) item.proxy;
 					if (row.getTableViewRowProxyItem() != null) {
 						sameView = row.getTableViewRowProxyItem().equals(convertView);
 					}
-					// TIMOB-24560: prevent duplicate TableViewRowProxyItem on Android N
-					if (Build.VERSION.SDK_INT > 23) {
-						ArrayList<Item> models = viewModel.getViewModel();
-						for (Item model : models) {
-							TableViewRowProxy proxy = (TableViewRowProxy) model.proxy;
-							if (proxy.getTableViewRowProxyItem().equals(convertView)) {
-								sameView = true;
-								v = null;
-								break;
-							}
-						}
+				}
+
+				// TIMOB-24560: prevent duplicate TableViewRowProxyItem on Android N
+				if (Build.VERSION.SDK_INT > 23) {
+					ArrayList<Item> models = viewModel.getViewModel();
+					if (models.contains(v.getRowData())) {
```

Comment:
You should check if "models" is null before using it.
It can be null if TableViewModel.release() gets called.

The rest looks fine. 👍